solidity/dashboard/src/sagas/liquidity-rewards.js

```diff
@@ -82,6 +94,62 @@ export function* watchFetchLiquidityRewardsData() {
   )
 }
 
+export function* watchLiquidityRewardsEarnedNotification() {
+  for (const [pairName, value] of Object.entries(LIQUIDITY_REWARD_PAIRS)) {
+    yield takeLatest(
+      `liquidity_rewards/${pairName}_liquidity_rewards_earned_notification`,
+      processLiquidityRewardEarnedNotification
+    )
+  }
+}
+
+function* processLiquidityRewardEarnedNotification(action) {
+  const contracts = yield getContractsContext()
+  const displayMessage = action.payload.displayMessage
+
+  const liquidityRewardPairName = action.payload.liquidityRewardPairName
+  const liquidityRewardPair = LIQUIDITY_REWARD_PAIRS[liquidityRewardPairName]
+
+  const LPRewardsContract = contracts[liquidityRewardPair.contractName]
+  const { liquidityRewards } = yield select()
+  const lastNotificationRewardAmount = new BigNumber(
+    liquidityRewards[
+      liquidityRewardPairName
+    ].liquidityRewardsEarnedNotification.lastNotificationRewardAmount
+  )
+  const currentReward = yield call(
+    fetchRewardBalance,
+    action.payload.address,
+    LPRewardsContract
+  )
+  // show the notification if the rewardBalance from LPRewardsContract is greater
+  // than the reward amount that was last time the notification was displayed
+  if (gt(currentReward, lastNotificationRewardAmount)) {
+    if (displayMessage) {
+      yield put(
+        showMessage({
+          type: messageType.REWARD,
+          title: `[${liquidityRewardPairName}] You've earned rewards!`,
+          sticky: true,
+          content: <ViewYourLiquidityBalance />,
+          classes: {
+            bannerDescription: "m-0",
+          },
+        })
+      )
```

Comment:
I was thinking that after showing the message we could call `fetchAllLiquidtyRewardsData()` so if the notification shows to the user while he is on Liquidity Page then he doesn't have to refresh to see the updated rewards. I'm not sure if it will be a good user experience though if he's on a Liquidity Page and suddenly the data starts fetching again. Thoughts?

Reply:
We can dispatch acion like `liquidity_rewards/${liquidityRewardPair.name}_reward_updated` and pass the `currentReward` value. Then we can update current reward value via reducer.